special-pages/pages/new-tab/app/widget-list/widget-config.provider.js

```diff
@@ -87,17 +97,21 @@ export function useVisibility() {
  * This wraps each widget and gives
  * @param {object} props
  * @param {WidgetConfigItem['id']} props.id - the current id key used for storage
- * @param {WidgetConfigItem['visibility']} props.visibility - the current id key used for storage
  * @param {number} props.index - the current id key used for storage
  * @param {import("preact").ComponentChild} props.children
  */
 export function WidgetVisibilityProvider(props) {
-    const { toggle } = useContext(WidgetConfigContext);
+    const { toggle, currentValues } = useContext(WidgetConfigContext);
+    const visibility = computed(() => {
+        const matchingConfig = currentValues.value.find((x) => x.id === props.id);
+        if (!matchingConfig) throw new Error('unreachable. Must find widget config via id: ' + props.id);
+        return matchingConfig.visibility;
+    });
```

Comment:
Compute the current visibility property based on 'currentValues' (which gets updated via subscriptions). Doing this with signals prevents VDOM diffing on unaffected sibling widgets